neo4j/internal/pool/pool.go

```diff
@@ -256,7 +287,10 @@ func (p *Pool) Borrow(ctx context.Context, serverNames []string, wait bool, bolt
 	case <-q.wakeup:
 		return q.conn, nil
 	case <-ctx.Done():
-		p.queueMut.Lock()
+		// TODO: provided ctx has reached deadline already - set some hardcoded timeout instead?
+		if !p.queueMut.TryLock(context.Background()) {
```

Comment:
this emulates the previous implementation which could possibly block forever... not sure that's the best thing to do, though.